### DISCUSSION TITLE
Kendala Keras LBB NN

### DISCUSSION THREAD

**Discussion by dimasrepo** [teamalgoritma/community#903] [Q&A] [answered]
Dear Team Algoritma,

Kendala penggunaan keras di R untuk LBB NN pada proses split data
berikut dokumentasinya

1. Eror
![image](https://github.com/user-attachments/assets/d34bf36e-0194-453b-b604-15c835163848)
2. Conda Env di R yang sudah di apply
![image](https://github.com/user-attachments/assets/011a6c46-7186-4abd-85b5-6a42c230e0a5)

**Comment by VictorNugraha** [comment]
Hi Pak @dimasrepo ,

Untuk kendala yang dihadapi sepertinya bukan karena koneksi antara R dengan Python.

Dugaan saya dari error yang terjadi mungkin dikarenakan bentukan data yang diinput kepada fungsi `to_categorical()`.

Mohon bantuannya dari Pak Dimas, untuk melampirkan print out dari variable train_x & train_y, agar kami bisa coba untuk telusuri lebih lanjut.

Terima kasih.

**Reply by VictorNugraha** [reply]
Sedikit tambahan, jika dirasa untuk variable train_x & train_y sudah sesuai.

Bapak bisa mencoba untuk mengganti versi Pythonnya menjadi 3.10 & versi TensorFlow ke 2.12.

Terima kasih.

**Reply by dimasrepo** [reply]
Hello mas, 
1. Bukan keras ya mas, saya kira keras karena saya rasa sudah sesuai datanya,  kira kira kenapa ya mas ? 

![image](https://github.com/user-attachments/assets/eefeedd8-e3b5-4353-91c9-57d5c97f6c37)
![image](https://github.com/user-attachments/assets/c660a6eb-b4d9-4277-8c52-592d0b1318ab)

2. Oke mas saya coba down version tapi nanti perlu buat env baru lagi ya di R nya untuk connect ke tensorflownya ? atau di env ikut berubah juga ya mas ? disininya 

![image](https://github.com/user-attachments/assets/e339031c-988c-435f-9d58-a789bff05cbd)

**Comment by ranydwi** [comment]
Dear Pak @dimasrepo , maaf baru menjawab kembali. Terima kasih untuk kesediannya memfollow-up pertanyaan ke Github Community. 

Untuk pertanyaan dan error yang Bapak alami sebetulnya pernah juga kejadian dengan student selama kelas. Hal ini terjadi karena adanya perbedaan version pada tensorflow dan juga python versionnya. 

![image](https://github.com/user-attachments/assets/307d537a-536f-46e2-afba-427605761e5d)

Silahkan ikuti pembuatan virtual environment baru pada [link berikut](https://docs.google.com/document/d/1mRba8rbUXFYD8VTD2iByeCbi9DhK_Q_b/edit?usp=sharing&ouid=106556574115290447377&rtpof=true&sd=true).

**Pastikan membuat environment baru dengan nama yang belum ada dengan Python version 3.10 dan Tensorflow 2.12**.

Note: Apabila Pak Dimas menggunakan Mac silahkan install Tensorflow dengan code sebagai berikut `conda install tensorflow==2.12` Semoga membantu.

**Comment by dimasrepo** [comment, marked as answer]
Dear Mbak Rany dan Mas Victor,

Ternyata tensorflownya aja yang perlu di downgrade, pythonnya sudah di 3.10. LBBnya juga sudah selesai besok saya kumpulkan. Terimakasih untuk bantuannya mbak dan mas.

Best Regards,

Dimas
